src/apm/helper.go

```diff
@@ -389,3 +400,27 @@ func createServiceInstanceId(namespaceName, podName, containerName string) strin
 	}
 	return serviceInstanceId
 }
+
+func addAnnotationToPodFromInstrumentationVersion(ctx context.Context, pod corev1.Pod, inst v1alpha2.Instrumentation) corev1.Pod {
```

Comment:
I am wondering if we should be using 
`func addAnnotationToPodFromInstrumentationVersion(ctx context.Context, pod *corev1.Pod, inst v1alpha2.Instrumentation) *corev1.Pod` to avoid all those pod copies and copy by errors that could happen